src/imars3dv2/filters/crop.py

```diff
@@ -0,0 +1,162 @@
+#!/usr/bin/env python3
+import numpy as np
+from scipy.ndimage import median_filter
+
+
+def crop(
+    arrays: np.ndarray,
+    crop_limit: tuple = (-1, -1, -1, -1),
+    border_pix: int = 10,
+    expand_ratio: float = 0.1,
+    rel_intensity_threshold_air_or_slit: float = 0.05,
+    rel_intensity_threshold_fov: float = 0.1,
+    rel_intensity_threshold_sample: float = 0.95,
+) -> np.ndarray:
+    """
+    Crop the image stack to provided limits or auto detected bound box.
+
+    Case 1: slits in, i.e I_inner >> I_outer
+    o-------------------------------------------------------
+    |        low                                           |
+    |     +++++++++++++++++top slit+++++++++++++++         |
+    |     +                                      +         |
+    |     +                                      +         |
+    |   left slit           FOV(high)         right slit   |
+    |     +                                      +         |
+    |     +                                      +         |
+    |     +++++++++++++++++bottom slit+++++++++++++        |
+    |                                                      |
+    --------------------------------------------------------
+
+    Case 2: slits out, i.e I_inner < I_outer
+    o-------------------------------------------------------
+    |                                                      |
+    |            *****************                         |
+    |             *    object   *           air            |
+    |              *   (low)   *           (high)          |
+    |               ***********                            |
+    |------------------------------------------------------|
+
+    Parameters
+    -----------
+    @param arrays: np.ndarray
```

Comment:
You don't need to annotate the types a second time here. Modern sphinx will read it from the function declaration.

Reply:
good to know, should I remove them now?